api/app/Models/PoolCandidate.php

```diff
@@ -541,6 +541,11 @@ public function computeAssessmentStatus()
             }
         }
 
+        // override to set current step as NULL if removed_at OR final_decision_at are set
+        if (! is_null($this->removed_at) || ! is_null($this->final_decision_at)) {
+            $currentStep = null;
+        }
```

Comment:
I felt it was best to continue to allow this method/location to be the source of truth for current step and just refer to it from more places. The RoD mutations are reversible so need to keep both directions in mind